src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -120,7 +120,12 @@ public DatabaseDialect create(AbstractConfig config) {
     }
   }
 
-  protected final Logger log = LoggerFactory.getLogger(getClass());
+  // This field should be private as it should only be used by this class and subclasses should
+  // instantiate their own loggers. However, it was originally declared as protected, and changing
+  // it to private now would break existing dialects that inherit from this class and rely on this
+  // field. So we leave it as protected now, but strongly recommend that all dialects use their own
+  // loggers (as all dialects in this code base do).
+  protected final Logger log = LoggerFactory.getLogger(GenericDatabaseDialect.class);
```

Comment:
hmm we could include `@Deprecated` but then it would mark all of our uses as deprecated too...
I guess we could also re-assign it to a private variable and ignore that one deprecation warning, and then rename all of the other accessors here.

Reply:
SGTM. I doubled the instantiation logic for `glog` instead of assigning `log` to it since the idiom for loggers is that they are `private static final` and `log` is not declared `static`, but apart from that I went with this change. Mind taking one more quick look?